client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/step-content.tsx

```diff
@@ -9,11 +13,47 @@ type StepContentProps = {
 	goToStep?: NavigationControls[ 'goToStep' ];
 };
 
-const StepContent = ( { siteSlug, submit, goNext, goToStep }: StepContentProps ) => (
-	<div className="launchpad__content">
-		<Sidebar siteSlug={ siteSlug } submit={ submit } goNext={ goNext } goToStep={ goToStep } />
-		<LaunchpadSitePreview siteSlug={ siteSlug } />
-	</div>
-);
+function sortByRegistrationDate( domainObjectA: ResponseDomain, domainObjectB: ResponseDomain ) {
+	const registrationDateA = domainObjectA.registrationDate;
+	const registrationDateB = domainObjectB.registrationDate;
+
+	if ( registrationDateA > registrationDateB ) {
+		return -1;
+	} else if ( registrationDateA < registrationDateB ) {
+		return 1;
+	}
+	return 1;
+}
+
+const StepContent = ( { submit, goNext, goToStep }: StepContentProps ) => {
+	const site = useSite();
+
+	const { data: allDomains = [] } = useGetDomainsQuery( site?.ID ?? null, {
+		retry: false,
+	} );
+
+	const domains = allDomains.map( createSiteDomainObject );
+
+	// Filter custom domains and sort by registration date.
+	// We want the first domain object to be the most recently registered domain
+	const nonWpcomDomains = domains
+		.filter( ( domain ) => ! domain.isWPCOMDomain )
+		.sort( sortByRegistrationDate );
+
+	const wpcomDomains = domains.filter( ( domain ) => domain.isWPCOMDomain );
```

Comment:
Would this also need to be sorted (like nonWpcomdomains) to ensure you're getting the most recent *.wordpress.com domain?

Reply:
Also wondering... would it be worth reducing the number of steps here by using something [lodash partition](https://lodash.com/docs/4.17.15#partition) to create the wp and non-wp domain lists at once? Can see it in use in Calypso [here](https://github.com/Automattic/wp-calypso/blob/trunk/client/state/selectors/get-sites.js).

No strong feelings on my side on this though.

Reply:
> Would this also need to be sorted (like nonWpcomdomains) to ensure you're getting the most recent *.wordpress.com domain?

Based on the /domains API, we are only receiving a single WP domain in the response.
p1665517041555449/1665513799.968999-slack-CHN6J22MP